src/pages/RegistrationPage.tsx

```diff
@@ -39,19 +43,47 @@ const RegistrationPage: React.FC = () => {
     }
 
     if (!hasError) {
-      console.log('First Name: ', firstName);
-      console.log('Last Name: ', lastName);
-      console.log('Email: ', email);
-      console.log('Password: ', password);
+      const requestBody = {
+        email,
+        password,
+        firstName: firstName || 'FName',
+        lastName: lastName || 'LName',
+        gender: 'DECLINE_TO_IDENTIFY'
+      };
+      try {
+        const response = await fetch('/auth/register', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(requestBody)
+        });
 
-      setFirstName('');
-      setLastName('');
-      setEmail('');
-      setPassword('');
-      setConfirmPassword('');
+        if (response.ok) {
+          window.location.href = '/home';
```

Comment:
Since you're already using react router in your app, it would be better to fully utilize it for navigation as well instead of just manipulating window.location.href. It has more refined functionalities and why use raw javascript unless you have to?